File: frontend/src/services/taskService.js
import API from "./api";

//API helpers for tasks
export const getAllTasks = () => API.get("/api/tasks");
export const getTasksByProject = (projectId) => API.get(`/api/tasks/${projectId}/tasks`);
export const getTasksByUser = (userId) => API.get(`/api/tasks/user/${userId}`);
export const createTask = (task) => API.post("/api/tasks",task);
export const updateTask = (id, updatedTask) => API.put(`/api/tasks/${id}`,updatedTask);
export const deleteTask = (id) => API.delete(`/api/tasks/${id}`);

File: EmployeeTaskManagementSystem/src/main/java/com/etms/EmployeeTaskManagementSystem/service/TaskService.java
package com.etms.EmployeeTaskManagementSystem.service;

import com.etms.EmployeeTaskManagementSystem.model.Task;
import com.etms.EmployeeTaskManagementSystem.model.TaskStatus;
import com.etms.EmployeeTaskManagementSystem.repository.TaskRepository;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;

@Service
public class TaskService {

    private final TaskRepository taskRepository;

    public TaskService(TaskRepository taskRepository) {
        this.taskRepository = taskRepository;
    }

    public Task createTask(Task task) {
        return taskRepository.save(task);
    }

    public List<Task> getAllTasks() {
        return taskRepository.findAll();
    }

    public Optional<Task> getTaskById(Long id) {
        return taskRepository.findById(id);
    }
    
    public List <Task> getTasksByProjectId(Long projectId) {
    	return taskRepository.findByProjectId(projectId);
    }
    
    public List<Task> getTasksByStatus(TaskStatus status) {
        return taskRepository.findByStatus(status);
    }

    public List<Task> getTasksByUser(Long userId) {
        return taskRepository.findByAssignedToId(userId);
    }


    public Task updateTask(Task task) {
        return taskRepository.save(task);
    }

    public void deleteTask(Long id) {
        taskRepository.deleteById(id);
    }
}
